server.js

```diff
@@ -2,49 +2,74 @@
 
 var request = require('request')
 var express = require('express')
+var xml2js = require('xml2js')
+var coberturaPath = '/jenkins/cobertura/'
+var jacocoPath = '/jenkins/jacoco/'
 
 var app = express()
+var badgeColor = function(cov) {
+	if (cov < 20) {
+		return 'red'
+	} else if (cov < 80) {
+		return 'yellow'
+	} else {
+		return 'brightgreen'
+	}
+}
+var getBadgeUrl = function(req, label, color) {
+	var badgeUrl = 'https://img.shields.io/badge/coverage-' + label + '-' + color + '.svg'
+	var style = req.param("style")
```

Comment:
Agreed

On Friday, October 2, 2015, Hutson Betts <notifications@github.com> wrote:

> In server.js
> <https://github.com/mnpk/jenkins-coverage-badge/pull/1#discussion_r41021698>
> :
>
> >
> >  var app = express()
> > +var badgeColor = function(cov) {
> > +	if (cov < 20) {
> > +		return 'red'
> > +	} else if (cov < 80) {
> > +		return 'yellow'
> > +	} else {
> > +		return 'brightgreen'
> > +	}
> > +}
> > +var getBadgeUrl = function(req, label, color) {
> > +	var badgeUrl = 'https://img.shields.io/badge/coverage-' + label + '-' + color + '.svg'
> > +	var style = req.param("style")
>
> Since the caller of getBadgeUrl is already familiar with the nature, and
> structure, of req, it doesn't sound unprecedented to have the caller pull
> out the style property..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mnpk/jenkins-coverage-badge/pull/1/files#r41021698>.
>